### PR TITLE
[CHW] API for external wallets, with empty handlers.

### PR DESCRIPTION
**Pull request by denisshevchenko**
## Description

This is the next PR for CHW (Cardano Hardware Wallet) support. It provides new API endpoints for external wallets.

## Linked issue

[CHW-54], [CHW-80], [CHW-106]

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: wallet-new/src/Cardano/Wallet/API/V1/Wallets.hs [hunk 1]
```diff
@@ -37,4 +37,16 @@ type API = Tags '["Wallets"] :>
...
+                   :> Summary "Check if this external wallet is presented in the node."
+                   :> PostCreated '[ValidJSON] (WalletResponse WalletAndTxHistory)
+    :<|> "external-wallets"
+                   :> Summary "Creates a new or restores an existing external wallet (mobile client or hardware wallet)."
+                   :> ReqBody '[ValidJSON] (New ExternalWallet)
+                   :> PostCreated '[ValidJSON] (WalletResponse Wallet)
+    :<|> "external-wallets"
+                   :> CaptureRootPK
+                   :> Summary "Deletes the given external wallet and all its accounts."
+                   :> DeleteNoContent '[ValidJSON] NoContent
```
Comment (KtorZ) [inline wallet-new/src/Cardano/Wallet/API/V1/Wallets.hs]: I'd avoid type aliases here `CaptureRootPK` especially if it's in another file :/ ... It increases the cognitive load for not-so-much added value in the end :/

File: wallet-new/src/Cardano/Wallet/API/V1/Types.hs [hunk 2]
```diff
@@ -602,6 +603,12 @@ instance BuildableSafeGen PublicKeyAsBase58 where
         %" }")
         ewalPublicKeyAsBase58
 
+instance FromHttpApiData PublicKeyAsBase58 where
+    parseQueryParam = Right . PublicKeyAsBase58Unsafe
```
Comment (KtorZ) [inline wallet-new/src/Cardano/Wallet/API/V1/Types.hs]: We don't have any smart constructor for this :thinking: ? So any piece of `Text` can be a `PublicKeyAsBase58` ?

File: wallet-new/src/Cardano/Wallet/API/V1/Handlers/Wallets.hs [hunk 3]
```diff
@@ -91,3 +93,21 @@ updateWallet pwl wid walletUpdateRequest = do
...
+                  -> NewExternalWallet
+                  -> Handler (WalletResponse Wallet)
+newExternalWallet _newExtWallet =
+    error "Cardano Hardware Wallet feature, new external wallet, unimplemented yet."
+
+deleteExternalWallet :: PassiveWalletLayer IO
+                     -> PublicKeyAsBase58
+                     -> Handler NoContent
+deleteExternalWallet _encodedRootPK =
+    error "Cardano Hardware Wallet feature, delete external wallet, unimplemented yet."
```
Comment (KtorZ) [inline wallet-new/src/Cardano/Wallet/API/V1/Handlers/Wallets.hs]: Do we have tickets for that? If not, can we create one and assign Squad 1 or Squad 2 to it :) ?

**Comment by denisshevchenko** [comment]
@parsonsmatt I made some changes. based on your comments.

**Pull request merged**